### ISSUE TITLE
Feature request: Different types of text justification

### ISSUE THREAD

**Issue by jilster** [kangax/fabric.js#4361]
With text justification, there are multiple ways to handle the alignment of the last line.

- Left justify: The last line is aligned left
- Right justify: The last line is aligned right
- Center justify: The last line is centered
- Full justify: The last line is justified

See: https://en.wikipedia.org/wiki/Typographic_alignment#Examples

fabric.js uses full justify, see this example and check the last line:
http://jsfiddle.net/rkuv7my0/1/

Would it be possible to make this a property in fabric.js?

**Comment by asturur** [comment]
Is not hard to do.
You can do it overriding a bit the code.
I have actually no time to make a prototype

**Comment by jilster** [comment]
I found a pull request covering the same question: https://github.com/kangax/fabric.js/pull/3896

Would you be interested in me continuing this? it would need some changes:
- Based off Fabric 2 beta instead of 1.x
- Include SVG export functionality
- Instead of overwriting default behaviour, adding new options for the textAlign property: 'justify' (current full justify, for backwards compatibility), 'justify-left', 'justify-right' and 'justify-center'

**Comment by asturur** [comment]
it mostly depends how much code is. if you want to give a shot do it.

keep it simple, slim, and check current fabricjs lint rules.

I cannot guarantee i ll merge it

**Comment by asturur** [comment]
there are 2 versions of _splitTextIntoLines, one for the text and one for the textbox.
Once splitted you have an array of lines that is called _textLines if i remember good, and they get rendered one by one.

**Comment by jilster** [comment]
Thanks, that pushed me in the right direction, i'm nearly done now

**Comment by jilster** [comment]
I decided to only add the 'justify-left' option. That and the default justify style are probably the only ones people will be using anyways. This way i was able to keep it very simple, only a couple of lines were added and changed and it is backwards compatible. There was apparently no need to change the toSVG function. I tested it and it already works.

If there are people that would like 'justify-right' or 'justify-center' i am willing to make those as well, but i think it would make the code more complex.

Here is my pull request: https://github.com/kangax/fabric.js/pull/4369
